packages/liveblocks-core/src/http-client.ts

```diff
@@ -1,11 +1,1400 @@
-import type { AuthValue } from "./auth-manager";
-import { HttpError } from "./lib/autoRetry";
-import type { JsonObject } from "./lib/Json";
+import type { AuthManager, AuthValue } from "./auth-manager";
+import {
+  convertToCommentData,
+  convertToCommentUserReaction,
+  convertToInboxNotificationData,
+  convertToInboxNotificationDeleteInfo,
+  convertToThreadData,
+  convertToThreadDeleteInfo,
+} from "./convert-plain-data";
+import { autoRetry, HttpError } from "./lib/autoRetry";
+import type { BatchStore } from "./lib/batch";
+import { Batch, createBatchStore } from "./lib/batch";
+import { chunk } from "./lib/chunk";
+import { createCommentId, createThreadId } from "./lib/createIds";
+import type { DateToString } from "./lib/DateToString";
+import type { Json, JsonObject } from "./lib/Json";
+import { objectToQuery } from "./lib/objectToQuery";
 import type { QueryParams, URLSafeString } from "./lib/url";
-import { urljoin } from "./lib/url";
+import { url, urljoin } from "./lib/url";
 import { raise } from "./lib/utils";
+import type { Permission } from "./protocol/AuthToken";
+import type { ClientMsg } from "./protocol/ClientMsg";
+import type {
+  BaseMetadata,
+  CommentAttachment,
+  CommentBody,
+  CommentData,
+  CommentDataPlain,
+  CommentLocalAttachment,
+  CommentUserReaction,
+  CommentUserReactionPlain,
+  QueryMetadata,
+  ThreadData,
+  ThreadDataPlain,
+  ThreadDeleteInfo,
+  ThreadDeleteInfoPlain,
+} from "./protocol/Comments";
+import type {
+  InboxNotificationData,
+  InboxNotificationDataPlain,
+  InboxNotificationDeleteInfo,
+  InboxNotificationDeleteInfoPlain,
+} from "./protocol/InboxNotifications";
+import type { IdTuple, SerializedCrdt } from "./protocol/SerializedCrdt";
+import type { HistoryVersion } from "./protocol/VersionHistory";
+import type { TextEditorType } from "./types/Others";
+import type { Patchable } from "./types/Patchable";
+import type { RoomNotificationSettings } from "./types/RoomNotificationSettings";
 import { PKG_VERSION } from "./version";
 
+export interface RoomHttpApi {
```

Comment:
Same as comment above. Ideally we'd bind `<M>` here at the group level, as there will be less room for errors that way, and we can benefit from the inference.